templates/FFV_form_factors.cpp.in

```diff
@@ -424,4 +752,430 @@ std::valarray<std::complex<double>> FFV_FFS<Fj, Fi, V, FA, FB, S>::value(
+   const typename field_indices<Fi>::type& indices_out,
+   context_base const& context,
+   bool discard_SM_contributions)
+{
+
+   static_assert(
+      std::is_same<FA, FB>::type::value,
+      "Internal fermions in the FFV_FFV instantiation must be of the same type."
+   );
+
+   // use P for external massless vector so as not to confuse with the internal vectors
+   using VertexFBarFjVBar = Vertex<typename V::lorentz_conjugate, typename FA::lorentz_conjugate, Fj>;
+   using VertexFiBarFV = Vertex<typename Fi::lorentz_conjugate, FB, V>;
+   using VertexFBarFPBar = Vertex<typename FB::lorentz_conjugate, FA, typename P::lorentz_conjugate>;
+
+   // masses of external fermions
+   const auto mj = context.mass<Fj>(indices_in);
+   const auto mi = context.mass<Fi>(indices_out);
+
+   // loop over all possible particle generations attached to both vertices
+   std::valarray<std::complex<double>> res {0.0, 0.0, 0.0, 0.0};
+   for (const auto& indexIn: index_range<VertexFBarFjVBar>()) {
+      for (const auto& indexOut: index_range<VertexFiBarFV>()) {
+           
+         // cycle if generations of external fermions are different then requested
+         const auto jFieldIndices = VertexFBarFjVBar::template indices_of_field<2>(indexIn);
+         const auto iFieldIndices = VertexFiBarFV::template indices_of_field<0>(indexOut);
+         if (jFieldIndices != indices_in || iFieldIndices != indices_out)
+            continue;
+
+         // match indices of the fermion in the loop
+         const auto fermionFieldIndicesIn = VertexFBarFjVBar::template indices_of_field<1>(indexIn);
+         const auto fermionFieldIndicesOut = VertexFiBarFV::template indices_of_field<1>(indexOut);
+         if (fermionFieldIndicesIn != fermionFieldIndicesOut)
+            continue;
+
+         // match indices of the vector in the loop
+         const auto vectorFieldIndicesIn = VertexFBarFjVBar::template indices_of_field<0>(indexIn);
+         const auto vectorFieldIndicesOut = VertexFiBarFV::template indices_of_field<2>(indexOut);
+         if (vectorFieldIndicesIn != vectorFieldIndicesOut)
+            continue;
+
+         // ignore contribution if both vector and fermion are SM particles
+         if (discard_SM_contributions) {
+            if (isSMField<FA>(fermionFieldIndicesIn) && isSMField<V>(vectorFieldIndicesIn)) {
+               continue;
+            }
+         }
+
+         const auto vertexIn = VertexFBarFjVBar::evaluate(indexIn, context);
+         const auto vertexOut = VertexFiBarFV::evaluate(indexOut, context);
+
+         const auto indexEmit = concatenate(fermionFieldIndicesIn, fermionFieldIndicesIn);
+         const auto vertexEmit = VertexFBarFPBar::evaluate(indexEmit, context);
+
+         const auto mV = context.mass<V>(vectorFieldIndicesIn);
+         const auto mF = context.mass<FA>(fermionFieldIndicesIn);
+         const auto x = Power2(mF/mV);
+
+	 std::complex<double> vector_boson_coupling {vertexEmit.left()};
```

Comment:
Please fix the indentation in this function.

Reply:
Done, thanks!